evil-types.el

```diff
@@ -370,6 +370,45 @@ If visual state is inactive then those values are nil."
   (when (evil-ex-p)
     (evil-ex-get-substitute-info evil-ex-argument t)))
 
+(evil-define-interactive-code "<d/>"
+  "Ex delete argument."
+  (when (evil-ex-p)
+    (evil-ex-get-delete-info evil-ex-argument)))
+
+(defun evil-ex-get-delete-info (string)
+  "Parse STRING as a :delete argument.
+Return a list (REGISTER COUNT)"
+  (let* (
+         (split-args (split-string (or string "")))
+         (arg-count (length split-args))
+         (arg0 (nth 0 split-args))
+         (arg1 (nth 1 split-args))
+         (number-regex "^[1-9][0-9]*$")
+         (register-regex "^[a-zA-Z]$")
+         (register nil)
+         (count nil))
+    (cond
+     ;; :delete REGISTER or :delete COUNT
+     ((eq arg-count 1)
```

Comment:
Use `=` for number comparisons.  The fact that `eq` works on integers (and characters, as they happen to be integers) is an implementation detail.